src/string-class.js

```diff
@@ -0,0 +1,191 @@
+   * @return {string}, a string with lower case characters
+   */
+  toLower() {
+    return this.replace(/[A-Z]/g, found =>
+      // makes use of ascii code
+      String.fromCharCode(found.charCodeAt(0) + 32)
+    );
+  },
+
+  /**
+   * ucFirst method converts the first character to upper case
+   * @return {string}, a string with it's initial character in upper case'
+   */
+  ucFirst() {
+    return this.replace(this.charAt(0), this.charAt(0).toUpper());
+  },
+
+  /**
+   * isQuestion method checks if a string is a question by
+   * checking if the last character is a question mark ?
+   * @return {boolean}; true if the string is a question and false otherwise
+   */
+  isQuestion() {
+    return /\?$/.test(this.trimSpace());
+  },
+
+  /**
+   * trimSpace method removes the trailing and preceeding empty
+   * spaces in a string
+   * @return {string}; trimmed string
+   */
+  trimSpace() {
+    return this.replace(/^\s+|\s+$/g, '');
+  },
+
+  /**
+   * words method gets all the words in a string
+   * @return {array}; an array of words in the string
+   */
+  words() {
+    const newValue = this.trimSpace().removeSpecialChars();
+    return newValue === '' ? [] : newValue.split(/\s+/);
+  },
+
+  /**
+   * removeSpecialChars method is used to replace non-alphanumeric characters
+   * including _ that are alone and replaces it with empty space
+   * @return {string}:
+   */
+  removeSpecialChars() {
+    return this.replace(/[^\w\s]|_/g, '');
+  },
+
+
+  /**
+   * wordCount method counts the number of words in a string
+   * @return {int}: number of words
+   */
+  wordCount() {
+    return this.words().length;
+  },
+
+  /**
+   * toCurrency method formats a number by putting commas in the right place
+   * @return {string}: string representing the currency format
+   */
+  toCurrency() {
+    const parts = this.split(/[.]/g),
+      length = parts[0].length;
+    let firstCommaPosition = length % 3;
+    firstCommaPosition = (firstCommaPosition === 0) ? 3 : firstCommaPosition;
+    /* eslint-disable prefer-template */
+    const regExpression =
+    new RegExp('^[0-9]{' + firstCommaPosition + '}|[0-9]{3}(?!$)', 'g');
+    /* eslint-enable prefer-template */
+    parts[0] = parts[0].replace(regExpression, number =>
+      `${number},`
+    );
+    return parts[1] ? `${parts[0]}.${parts[1]}` : `${parts[0]}.00`;
+  },
+
+  /**
+   * fromCurrency format converts a string in the currency format by
+   * removing commas
+   * @return {string}: representing the pure digit format
+   */
+  fromCurrency() {
+    return this.replace(/,+/g, '');
+  },
+
+  /**
+   * inverseCase inverts the case of a character, a lower case character
+   * becomes upper case
+   * @return {string}: a string with inverted case
+   */
+  inverseCase() {
+    return this.replace(/[a-zA-Z]/g, (match) => {
+      const answer = /[A-Z]/.test(match) ? match.toLower() : match.toUpper();
+      return answer;
```

Comment:
I've removed the redundant variable. However, one of the eslint rule `no-confusing-arrow` doesn't agree with conditional return statements